src/instrumentations/http.ts

```diff
@@ -14,13 +14,106 @@
  * limitations under the License.
  */
 
-import { Options } from '../options';
-import { ServerResponse } from 'http';
+import { Options, CaptureHttpUriParameters } from '../options';
+import { IncomingMessage, ServerResponse } from 'http';
 import {
   HttpInstrumentationConfig,
   HttpResponseCustomAttributeFunction,
+  HttpRequestCustomAttributeFunction,
 } from '@opentelemetry/instrumentation-http';
-import { isSpanContextValid, TraceFlags } from '@opentelemetry/api';
+import { diag, isSpanContextValid, TraceFlags } from '@opentelemetry/api';
+import { Span } from '@opentelemetry/api';
+import * as Url from 'url';
+
+type IncomingHttpRequestHook = (span: Span, request: IncomingMessage) => void;
+
+function shouldAddRequestHook(options: Options): boolean {
```

Comment:
I guess we're doing this to skip early? However, it decouples the actual hook creation from this check so we essentially need to do it again the hook creation function.

I'd suggest just use one place for the hook creation and return early there for optimization.
Also have a case which applies for invalid input.

Reply:
It's there just to avoid attaching a request hook in the first place. The other check is when attaching the internal request processor.